_posts/2017-04-30-React-Native-for-iOS-devs.markdown

```diff
@@ -0,0 +1,248 @@
+- Provides a tonne of primitives to allow writing declaritive animations
+
+Places where React Native hasn't fit for us
+
+- So far, nowhere. Lols. Can't be true overall though.
+- Something like our Kiosk app isn't an amazing fit
+- Storyboards is a great abstraction for this kind of app, it has lots of obviously connected screens
+- However, if we could have re-used our components from another app really easily, it'd probably have been easier to write, significantly easier to maintain. We'd also have considerably more contributions from people outside of the core two developers who worked on it.
+
+When to choose React Native?
+
+- Maybe update this post WWDC?
+- React Native provides a cross platform API, and so it can fall into a usual watered down version of the API it abstracts. 
+- Lots of apps don't bother with all the hard stuff though.
+- For me, any app whose main job is to take an API and turn it into data should probably be a react native app.
+- Apple gives a lot of polish on it's dev tools, but at the price of a closed system.
+
+
+Now that I am reasonably proficient with the trade-offs, if I would use React Native
+
+- An app whose sole purpose is to consume an external resource: e.g. an API.
+- An app that doesn't easily fit into a single storyboard.
+- Any app where Android is a hard requirement, because we don't have an android team, we can get it 70% of the way before we have to have someone with deep skills on the team.
+ 
+I wouldn't use React Native if I were writing:
+
+- A non-traditional app: e.g. a watchOS/iMessage/macOS app.
+
+These are not platforms that are well tested, and kept up to date by the ever-changing node world. You can do it, but you will definitely have to get your hands dirty in the React Native source-code for the platform..
+
+- An encrypted chat client
+- Apps for a bank
+- Apps that interact with life-critical systems like health
+
+
+Not because you are shipping the source code for your app inside the app, that's never stopped websites handling the above. The bridging between runtimes is leaky, and is a great place vector for attacking your application. 
+
+- Working in RN gives you the chance to have more of your app easier to de-silo your mobile engineering teams, making it simpler to write code across all teams because you have  with
+
+
+
+Brownfield
+
+If you're thinking of adding RN to an existing app, first read [on emission][]. We think of our RN to be a series of components which are consumed by our app as a CocoaPod.
+
+Greenfield
+
+I'd probably start with a CRNA app, it's a good starting point. I feel safe that I can eject out of the environment provided when the app becomes complex enough to warrant native code
+
+I would probably start with this boiler-plate, but I am a domain expert now. In order to feel comfortable with it, you'll need to be comfortable with 
+
+* TypeScript
+* Babel
```

Comment:
Researching this turned into a 20 minute exercise. But the best template is the one from Microsoft: https://github.com/Microsoft/TypeScript-React-Native-Starter

Which does the 2 step process of TS -> Babel

> Currently the way that React Native operates is that the React Native Packager runs .js files through Babel and bundles the output for the device. At the moment, there is no easy way to configure the packager to run directly on .tsx files, but given TypeScript's emit speed, it's very reasonable to have React Native pick up TypeScript's output.

So might still be worth keeping around